source/content/guides/drupal-9-migration/06-build-tools-to-d9-build-tools.md

```diff
@@ -0,0 +1,242 @@
+
+`terminus drush {site name}.{integration env} pm-enable upgrade_status --yes`
+
+You can get a one-time login to your site with the following command: 
+`terminus drush {site name}.{integration env} uli admin`
+
+Log into your site as admin and take a look under reports at "UPGRADE STATUS". Any modules Upgrade Status says are incompatible will need to be updated in the next few steps. Take note of the versions "UPGRADE STATUS" recommends. If your module is compatible it will need to be removed from the composer file.
+
+## Custom Module Code:
+
+Custom module code is outside the scope of this document. See drupal.org for getting your custom code updated with the new version numbers and any code deprecations.
+
+* Temporarily add write access to protected files and directories:
+
+```bash
+		chmod 777 web/sites/default
+		find web/sites/default -name "*settings.php" \
+		  -exec chmod 777 {} \;
+		find web/sites/default -name "*services.yml" \
+		  -exec chmod 777 {} \;
+```
+
+## Heads turn as the star of our show makes an entrance: **CORE**
+
+```bash
+
+composer remove drupal/config_installer --no-update
+composer require drupal/core-recommended:^9 \
+   drupal/core-composer-scaffold:^9 \
+   drupal/core-project-message:^9  \
+   drush/drush:^10 \
+   -W --no-update
+
+composer require phpunit/phpunit:^9 \
+	 behat/behat:^3 \
+	 drupal/drupal-extension:^4 \
+	 --no-update -W --dev
+
+# If you have drupal/core-dev installed.
+
+composer require drupal/core-dev:^9 \
+	 --dev -W --no-update
+
+```
+
+If you have any obsolete modules uncovered by "UPGRADE STATUS" update them here using the '**--no-update -W**' switch because that will keep composer from the process of figuring out the module solution (which is the thing that takes so long in a composer run) until the very end and update all the dependencies along with.
+
+When you're done updating modules, run
+
+`composer update -W --optimize-autoloader --prefer-dist`
+
+If you get an error, you probably haven't weeded out all the non-d9 module and theme updates and will need to have another look at the "Upgrade status" report in your integration environment.
+
+Once the update/install runs clean without errors:
+
+* Edit your composer file and change the following:
+
+```composer.json
+[...]
+  "scripts": {
+    "build-assets": [
+      "@prepare-for-pantheon",
+      "composer install --optimize-autoloader"
+    ],
+[...]
+```
+
+TO:
+
+```composer.json
+[...]
+  "scripts": {
+    "build-assets": [
+      "@prepare-for-pantheon",
+      "composer install --optimize-autoloader --no-dev"
+    ],
+[...]
+```
+
+That way, when the build arrives in production it arrives without all the dev dependencies.
+
+```
+git add composer.json composer.lock pantheon.yml
+git commit -m 'upgrade core to d9'
+git push origin d9-upgrade-2021
+```
+
+1. Review the site, then proceed to launch using the [Pantheon Relauch](/relaunch) documentation.
```

Comment:
This is a migrate-in-place, so there shouldn't be any need to do a relaunch at this point.

At this point the Status Report page should be reporting the correct version of the MariaDb server. The Upgrade Status page should also be correct vis-a-vis the database version, but note that once you upgrade to Drupal 9, the Upgrade Status page starts reporting your readiness to upgrade to *DRUPAL TEN*.